### ISSUE TITLE
Messages with QoS levels 1 and 2 are not "echoed back" to this client

### ISSUE THREAD

**Issue by cbenhagen** [shamblett/mqtt_client#33]
Running the example script results in the following output:
```
EXAMPLE::Mosquitto client connecting....
EXAMPLE::Mosquitto client connected
EXAMPLE::Publishing our topic
EXAMPLE::Sleeping....
```

Changing line 104 to
```dart
client.publishMessage(pubTopic, MqttQos.atMostOnce, builder.payload);
```

```
EXAMPLE::Mosquitto client connecting....
EXAMPLE::Mosquitto client connected
EXAMPLE::Publishing our topic
EXAMPLE::Sleeping....
EXAMPLE::Change notification:: topic is <Dart/Mqtt_client/testtopic>, payload is <-- Hello from mqtt_client -->
```

Also receiving messages sent with 
```bash
mosquitto_pub -t "Dart/Mqtt_client/testtopic" -m "hello world!" -q 0
```
works. But not with `-q 1` or `-q 2`

```
mqtt_client versions: 3.1.0 and 3.0.0
Dart VM version: 2.0.0 (Fri Aug 3 10:53:23 2018 +0200) on "macos_x64"
Mosquitto 1.5.1
```

**Comment by shamblett** [comment]
Ok I'll have a look at this.

**Comment by cbenhagen** [comment]
I did some more tests. Messages with all QoS levels are received as expected by other subscribers. Only those with QoS 0 are "echoed back" to this client. This is the log from two messages. First with QoS 1 the second with QoS 0:

```
I/flutter (10989): 2018-09-10 17:38:36.561699 -- MqttConnectionHandler::sendMessage - MQTTMessage of type MqttMessageType.publish
I/flutter (10989): Header: MessageType = MqttMessageType.publish, Duplicate = false, Retain = false, Qos = MqttQos.atLeastOnce, Size = 0
I/flutter (10989): Publish Variable Header: TopicName={asdf}, MessageIdentifier={15}, VH Length={0}
I/flutter (10989): Payload: {4 bytes={<81><111><83><49>
I/flutter (10989): 2018-09-10 17:38:36.570282 -- MqttConnection::_onData
I/flutter (10989): 2018-09-10 17:38:36.571590 -- MqttConnection::_onData - message received MQTTMessage of type MqttMessageType.publishAck
I/flutter (10989): Header: MessageType = MqttMessageType.publishAck, Duplicate = false, Retain = false, Qos = MqttQos.atMostOnce, Size = 2
I/flutter (10989): PublishAck Variable Header: MessageIdentifier={15}
I/flutter (10989): 2018-09-10 17:38:36.571799 -- MqttConnection::_onData - message processed
I/flutter (10989): 2018-09-10 17:38:50.735957 -- MqttConnectionHandler::sendMessage - MQTTMessage of type MqttMessageType.publish
I/flutter (10989): Header: MessageType = MqttMessageType.publish, Duplicate = false, Retain = false, Qos = MqttQos.atMostOnce, Size = 0
I/flutter (10989): Publish Variable Header: TopicName={asdf}, MessageIdentifier={16}, VH Length={0}
I/flutter (10989): Payload: {4 bytes={<81><111><83><48>
I/flutter (10989): 2018-09-10 17:38:50.745298 -- MqttConnection::_onData
I/flutter (10989): 2018-09-10 17:38:50.747420 -- MqttConnection::_onData - message received MQTTMessage of type MqttMessageType.publish
I/flutter (10989): Header: MessageType = MqttMessageType.publish, Duplicate = false, Retain = false, Qos = MqttQos.atMostOnce, Size = 10
I/flutter (10989): Publish Variable Header: TopicName={asdf}, MessageIdentifier={0}, VH Length={6}
I/flutter (10989): Payload: {4 bytes={<81><111><83><48>
I/flutter (10989): 2018-09-10 17:38:50.747992 -- MqttConnection::_onData - message processed
I/flutter (10989): 1
I/flutter (10989): MQTT message: topic is <asdf>, payload is <-- QoS0 -->
I/flutter (10989): ConnectionState.connected
```

I am testing this with the new flutter example btw. It needs some polish but you can expect a PR in the next days. Here a little teaser:

![screenshot_1536594262](https://user-images.githubusercontent.com/814785/45308378-9e594980-b521-11e8-9f93-61854e677fe3.png)
![screenshot_1536594325](https://user-images.githubusercontent.com/814785/45308379-9e594980-b521-11e8-8f40-b4de4a5aaf37.png)
![screenshot_1536594343](https://user-images.githubusercontent.com/814785/45308380-9e594980-b521-11e8-8553-08e3cc1b1b44.png)

**Comment by shamblett** [comment]
Its a bit odd, looking at the mosquito issues there is this one https://github.com/eclipse/mosquitto/issues/331 that sorts of hints at this, I'll do some more testing with qos 2 and maybe another broker. Thanks for the example work.

**Comment by shamblett** [comment]
Possibly fixed on issue 38, please update to version 3.3.1 and retest.

**Comment by cbenhagen** [comment]
Yes this has also been fixed. Thanks!
